File: src/Java/uk/co/armedpineapple/cth/FileDetails.java
package uk.co.armedpineapple.cth;

import java.util.Date;

public class FileDetails implements Comparable<FileDetails> {

    private final Date   lastModified;
    private final String fileName;


    private final String directory;

    public FileDetails(String filename, String directory, Date lastModified) {
        this.fileName = filename;
        this.lastModified = lastModified;
        this.directory = directory;
    }


    public String getDirectory() {
        return directory;
    }

    public Date getLastModified() {
        return lastModified;
    }

    public String getFileName() {
        return fileName;
    }

    @Override
    public int compareTo(FileDetails another) {
        if (lastModified.equals(another.getLastModified())) {
            return 0;
        }

        return lastModified.after(another.getLastModified()) ? 1 : -1;

    }
}

File: src/Java/uk/co/armedpineapple/cth/FileDetails.kt
package uk.co.armedpineapple.cth

import java.util.*

data class FileDetails(val fileName: String, val directory: String, private val lastModified: Date) : Comparable<FileDetails> {

    override fun compareTo(other: FileDetails): Int {
        if (lastModified == other.lastModified) {
            return 0
        }

        return if (lastModified.after(other.lastModified)) 1 else -1

    }
}
